### PR TITLE
MGDOBR-723: OUIA in the  component

### PR DESCRIPTION
**Pull request by sjamboro**
https://issues.redhat.com/browse/MGDOBR-723

### CODE REVIEW

File: src/app/Processor/ProcessorDetail/ProcessorDetail.tsx [hunk 1]
```diff
@@ -61,7 +61,12 @@ const ProcessorDetail = (props: ProcessorDetailProps): JSX.Element => {
           <Stack hasGutter={true}>
             <StackItem>
               <TextContent>
-                <Text component={TextVariants.h2}>{t("processor.source")}</Text>
+                <Text
+                  component={TextVariants.h2}
+                  ouiaId="processor-source-section"
```
Comment (jomarko) [inline src/app/Processor/ProcessorDetail/ProcessorDetail.tsx:66]: To be consistent with the rest of processor details, I think we should use just `source-section`.

**Review by jomarko — COMMENTED** [review]
Thank you @sjamboro . One comment from me.

**Review by vpellegrino — APPROVED** [review]
LGTM, thank you @sjamboro

**Review by jomarko — APPROVED** [review]
Thank you @sjamboro

**Review by kelvah — APPROVED** [review]
LGTM. Thank you @sjamboro. Left a comment/question.

File: src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx [hunk 2]
```diff
@@ -91,6 +92,7 @@ const FiltersEdit = (props: FiltersEditProps): JSX.Element => {
                     <TextInput
                       type="text"
                       id={`filter-key-${index}`}
+                      ouiaId="filter-key"
```
[the remark this answers is not in the document]
Comment (jomarko) [reply src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx:95]: @kelvah good question. 

You are partially correct, if we speak about html elements, that have just `data-ouia-component-id` without `data-ouia-component-type` specified.

However, the `data-ouia-component-id` doesn't need to be unique across a whole document if html elements belong to different  `data-ouia-component-type`s.

example bellow, is fine, `id`s are same, but `type`s are not.
```html
<div data-ouia-component-type='modal' data-ouia-component-id='submit'>
  <button data-ouia-component-type='modal-button' data-ouia-component-id='submit'>Submit<>
</div>
```

[the remark this answers is not in the document]
Comment (kelvah) [reply src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx:95]: Got it. Thanks @jomarko. :+1:

**Review by sthundat — APPROVED** [review]

File: src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sjamboro) [reply src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx:95]: @kelvah from specification:
`it MUST be unique within the surrounding context of the component (context may be a surrounding component or collection)` 

that means that all `td` in the same row must be unique. All `tr` in the same table must be unique.

[the remark this answers is not in the document]
Comment (kelvah) [reply src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx:95]: Thank you @sjamboro for the clarification. :+1:

**Comment by sjamboro** [comment]
Merge conflict resolved, branch is on the top of the main, build & quality checks are green.

**Pull request merged**